src/coreclr/pal/src/arch/arm64/context2.S

```diff
@@ -154,6 +184,31 @@ LOCAL_LABEL(Restore_CONTEXT_FLOATING_POINT):
     // since we potentially clobber x0 below, we'll bank it in x16
     mov x16, x0
 
+    ldr w17, [x16, CONTEXT_XSTATEFEATURESMASK_OFFSET]
```

Comment:
It should check the CONTEXT_XSTATE in CONTEXT_ContextFlags first and check the features mask only if the CONTEXT_XSTATE is set.